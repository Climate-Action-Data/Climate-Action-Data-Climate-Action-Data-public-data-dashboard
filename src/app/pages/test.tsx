'use client'
import { type NextPage } from "next";

const Test: NextPage = () => {
    return (
        <div>
            <h1>Hello</h1>
            <p>data</p>
        </div>
    )
}

export default Test