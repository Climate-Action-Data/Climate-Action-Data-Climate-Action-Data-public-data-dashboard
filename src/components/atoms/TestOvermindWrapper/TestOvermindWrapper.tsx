import { createOvermind } from 'overmind'
import { Provider } from 'overmind-react'
import { config } from '@/overmind'
import { PropsWithChildren } from 'react'

const overmind = createOvermind(config)

export const TestOvermindWrapper = (props: PropsWithChildren) => {
  return <Provider value={overmind}>{props.children}</Provider>
}
