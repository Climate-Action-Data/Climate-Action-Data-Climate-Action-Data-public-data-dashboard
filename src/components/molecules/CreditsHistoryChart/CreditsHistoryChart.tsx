import { FC } from 'react'

import { useAppState } from '@/overmind'
import CreditHistoryLineChart from '@/components/molecules/CreditsHistoryChart/CreditsHistoryLineChart'

const CreditsHistoryChart: FC = () => {
  const { filteredCreditsHistory } = useAppState().creditsHistory

  const chartDataPoints = filteredCreditsHistory?.chartData

  if (chartDataPoints) {
    return <CreditHistoryLineChart data={chartDataPoints} />
  }
  return <></>
}
export default CreditsHistoryChart
