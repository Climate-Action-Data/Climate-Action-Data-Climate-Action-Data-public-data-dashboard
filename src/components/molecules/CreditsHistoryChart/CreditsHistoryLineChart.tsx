import { CreditsHistoryChartData } from '@/@types/State'
import { ResponsiveLine } from '@nivo/line'

const CreditHistoryLineChart = (props: { data: CreditsHistoryChartData[] }) => {
  const format = new Intl.DateTimeFormat(`en-SG`, { month: `short`, year: `2-digit` }).format
  return (
    <ResponsiveLine
      colors={[`#24BD63`, `#125E32`]}
      data={props.data}
      margin={{ top: 50, right: 50, bottom: 50, left: 50 }}
      xScale={{ type: `point` }}
      yScale={{
        type: `linear`,
        max: `auto`,
        stacked: false,
        reverse: false,
      }}
      yFormat=" >-.2s"
      xFormat={(value) => format(new Date(value)).toLocaleUpperCase()}
      axisTop={null}
      axisRight={null}
      axisBottom={{
        tickSize: 0,
        tickPadding: 10,
        tickRotation: 0,
        legend: ``,
        legendOffset: 36,
        legendPosition: `middle`,
        format: (value) => format(new Date(value)).toLocaleUpperCase(),
      }}
      axisLeft={{
        tickSize: 0,
        tickPadding: 11,
        tickRotation: 0,
        legend: ``,
        legendOffset: -40,
        legendPosition: `middle`,
        format: ` >-.1s`,
        tickValues: 5,
      }}
      enableGridX={true}
      gridXValues={[0]}
      enableGridY={true}
      gridYValues={[0]}
      pointColor={{ from: `color`, modifiers: [] }}
      pointBorderColor={{ from: `serieColor` }}
      pointLabelYOffset={-12}
      useMesh={true}
      legends={[]}
      animate={false}
    />
  )
}
export default CreditHistoryLineChart
