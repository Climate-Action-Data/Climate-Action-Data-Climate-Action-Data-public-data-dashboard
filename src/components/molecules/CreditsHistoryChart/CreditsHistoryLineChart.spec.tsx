import { render } from '@testing-library/react'

import { TestOvermindWrapper } from '@/test/TestOvermindWrapper'
import { MockData } from '@/test/TestOvermindMockData'
import CreditHistoryLineChart from '@/components/molecules/CreditsHistoryChart/CreditsHistoryLineChart'
import { calculateTickCount } from '@/components/molecules/CreditsHistoryChart/CreditsHistoryLineChart'

describe(`CreditsHistoryChart`, () => {
  test(`renders loading state when creditsHistoryData is undefined`, () => {
    const { container } = render(
      <TestOvermindWrapper>
        <CreditHistoryLineChart data={MockData.CREDIT_HISTORY_CHART_DATA} />
      </TestOvermindWrapper>,
    )
    expect(container).toMatchSnapshot()
  })
})

describe(`calculateTickCount`, () => {
  it(`correctly calculates the tick count when dataLength is less than maxTickCount`, () => {
    const dataLength = 5
    const maxTickCount = 12
    const result = calculateTickCount(dataLength, maxTickCount)
    expect(result).toBe(dataLength)
  })

  it(`correctly calculates the tick count when dataLength is greater than maxTickCount`, () => {
    const dataLength = 15
    const maxTickCount = 12
    const result = calculateTickCount(dataLength, maxTickCount)
    expect(result).toBe(maxTickCount)
  })

  it(`correctly calculates the tick count when dataLength is equal to maxTickCount`, () => {
    const dataLength = 12
    const maxTickCount = 12
    const result = calculateTickCount(dataLength, maxTickCount)
    expect(result).toBe(maxTickCount)
  })
})
