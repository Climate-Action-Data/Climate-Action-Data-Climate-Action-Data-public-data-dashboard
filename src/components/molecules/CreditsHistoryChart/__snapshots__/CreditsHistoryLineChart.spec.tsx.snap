// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditsHistoryChart renders loading state when creditsHistoryData is undefined 1`] = `
<div>
  <div
    style="width: 100%; height: 100%;"
  />
</div>
`;
